cpp/src/db/db_object.cuh

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cugraph.h>
+#include <vector>
+#include <map>
+#include "utilities/graph_utils.cuh"
+
+namespace cugraph {
+  /**
+   * Class for representing an entry in a pattern, which may either be a variable or constant value
+   */
+  template <typename idx_t>
+  class db_pattern_entry {
+    bool is_var;
+    idx_t constantValue;
+    std::string variableName;
+  public:
```

Comment:
So here, idx_t constantValue is used for vertex indices and std::string variableName is used for label/property names we want to match? Adding comments explaining how this maps to cypher queries will be really helpful in understanding this code.

Reply:
No, in a pattern you have entries that are either a variable or a constant. In the case you are referring to the label or property names would be constants and represented by an idx_t (since all constants are dictionary-encoded to ids before getting stored in the table.

Reply:
I have added some more comments explaining this further.